client-js/typings/whatwg-fetch/whatwg-fetch.d.ts

```diff
@@ -0,0 +1,98 @@
+﻿// Type definitions for fetch API
+// Project: https://github.com/github/fetch
+// Definitions by: Ryan Graham <https://github.com/ryan-codingintrigue>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare class Request extends Body {
+	constructor(input: string|Request, init?:RequestInit);
+	method: string;
+	url: string;
+	headers: Headers;
+	context: RequestContext;
+	referrer: string;
+	mode: RequestMode;
+	redirect: RequestRedirect;
+	credentials: RequestCredentials;
+	cache: RequestCache;
+}
+
+interface RequestInit {
+	method?: string;
+	headers?: HeaderInit|{ [index: string]: string };
+	body?: BodyInit;
+	mode?: RequestMode;
+	redirect?: RequestRedirect;
+	credentials?: RequestCredentials;
+	cache?: RequestCache;
+}
+
+type RequestContext =
+	"audio" | "beacon" | "cspreport" | "download" | "embed" |
+	"eventsource" | "favicon" | "fetch" | "font" | "form" | "frame" |
+	"hyperlink" | "iframe" | "image" | "imageset" | "import" |
+	"internal" | "location" | "manifest" | "object" | "ping" | "plugin" |
+	"prefetch" | "script" | "serviceworker" | "sharedworker" |
+	"subresource" | "style" | "track" | "video" | "worker" |
+	"xmlhttprequest" | "xslt";
+type RequestMode = "same-origin" | "no-cors" | "cors";
+type RequestRedirect = "follow" | "error" | "manual";
+type RequestCredentials = "omit" | "same-origin" | "include";
+type RequestCache =
+	"default" | "no-store" | "reload" | "no-cache" |
+	"force-cache" | "only-if-cached";
+
+declare interface HeadersMap {
+	[index: string]: string;
+}
+
+declare class Headers {
+	constructor(headers?:Headers|HeadersMap)
+	append(name: string, value: string): void;
+	delete(name: string):void;
+	get(name: string): string;
+	getAll(name: string): Array<string>;
```

Comment:
This is straight from the DefinitelyTyped repo, so I'm going to leave it as is (this one didn't require any changes to work with es6 modules since it just polyfills a global function).